tester/src/test/scala/spinal/tester/scalatest/SpinalSimWishboneSlaveFactoryTester.scala

```diff
@@ -43,55 +46,66 @@ class SpinalSimWishboneSlaveFactoryTester extends SpinalAnyFunSuite{
+        (bus, transaction, we) => {
+          scoreboard.pushDut(transaction.copy(data = bus.DAT_MISO.toBigInt))
+        }
+      )
+
+      // Make sure that all configurations exhibit proper latencies.
+      val expectedResponseTime = if(regFeedback) 10 else 0
+      val busStatus = WishboneStatus(dut.io.bus)
+      var responseTime : Option[Long] = None
+      dut.clockDomain.onSamplings({
+        if(busStatus.isCycle) {
+          // Start the clock as soon as we see a request
+          if (busStatus.masterHasRequest && responseTime.isEmpty) {
+            responseTime = Some(simTime())
+          }
+
+          // Measure until ACK time.
+          if(dut.io.bus.ACK.toBoolean) {
+            val elapsedTime = simTime() - responseTime.get
+            //assert(expectedResponseTime == elapsedTime)
+
+            // For the pipeline case; if STB is assert it means we got a request now
+            responseTime = if(conf.isPipelined && dut.io.bus.STB.toBoolean && regFeedback) Some(simTime()) else None
+          }
+        }
+      })
 
       val dri = new WishboneDriver(dut.io.bus, dut.clockDomain)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc), WishboneTransaction(0)), we = false)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc, 200), WishboneTransaction(0, 2)), we = true)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc), WishboneTransaction(0)), we = false)
+      dri.drive(scala.collection.immutable.Seq(WishboneTransaction(21*wordInc, 0)), we = false)
+      dri.drive(scala.collection.immutable.Seq(WishboneTransaction(21*wordInc, 42)), we = true)
+      dri.drive(scala.collection.immutable.Seq(WishboneTransaction(21*wordInc)), we = false)
 
       scoreboard.checkEmptyness()
-//
-//      for(repeat <- 0 until 100){
-//        driver.drive(for(x <- 1 to 10) yield WishboneTransaction(address=10).randomizeData(dut.io.bus.config.dataWidth), we = true)
-//        dut.clockDomain.waitSampling(10)
-//        driver.drive(for(x <- 1 to 10) yield WishboneTransaction(address=10), we = false)
-//        dut.clockDomain.waitSampling(10)
-//      }
-//
 
     }
   }
-
-  test("classic"){
-    val conf = WishboneConfig(8,8)
-    testBus(conf,description="classic")
-  }
-
-  test("classic32"){
-    val conf = WishboneConfig(32, 32)
-    testBus(conf, description="classic32")
-  }
-
-  test("classic32_byte"){
-    val conf = WishboneConfig(32, 32, addressGranularity = AddressGranularity.BYTE)
-    testBus(conf, description="classic32_byte")
-  }
-
-  test("classic32_word"){
-    val conf = WishboneConfig(32, 32, addressGranularity = AddressGranularity.WORD)
-    testBus(conf, description="classic32_word")
+  for(regFeedback <- Seq(true, false)) {
+    for(pipelined <- Seq(true, false)) {
+      for(granularity <- Seq(AddressGranularity.BYTE, AddressGranularity.WORD, AddressGranularity.UNSPECIFIED)) {
+        for(dataWith <- Seq(8, 16, 32)) {
```

Comment:
same question as above to the widths: if only the lower 8 bit are really checked, we could remove the other widths

Reply:
I kept the different data widths where I thought it made sense to test it. I moved the sequence generation stuff to a library call which would excecise the full data width